radis/io/kurucz.py

```diff
@@ -0,0 +1,170 @@
+def fetch_kurucz(
+    molecule,
+    local_databases=None,
+    databank_name="Kurucz-{molecule}",
+    isotope=None,
+    load_wavenum_min=None,
+    load_wavenum_max=None,
+    columns=None,
+    cache=True,
+    verbose=True,
+    chunksize=100000,
+    clean_cache_files=True,
+    return_local_path=False,
+    engine="default",
+    output="pandas",
+    parallel=True,
+):
+    """
+    Note: if a registered entry already exists and `radis.config["ALLOW_OVERWRITE"]` is True:
+    - if any situation arises where the databank needs to be re-downloaded, the possible urls are attempted in their usual order of preference, as if the databank hadn't been registered, rather than directly re-downloading from the same url that was previously registered, in case e.g. a new linelist has been uploaded since the databank was previously registered
+    - If no partition function file is registered, e.g because one wasn't available server-side when the databank was last registered, an attempt is still made again to download it, to account for e.g. the case where one has since been uploaded
+    """
+
+    #largely based on :py:func:`~radis.io.fetch_geisa`
+    if r"{molecule}" in databank_name:
+        databank_name = databank_name.format(**{"molecule": molecule})
+
+    if local_databases is None:
+
+        local_databases = join(radis.config["DEFAULT_DOWNLOAD_PATH"], "kurucz")
+    local_databases = abspath(expanduser(local_databases))
+
+    ldb = KuruczDatabaseManager(
+        databank_name,
+        molecule=molecule,
+        local_databases=local_databases,
+        verbose=verbose,
+        parallel=parallel,
+        engine=engine,
+    )
+
+    local_files, urlnames = [], []
+    pf_path = []
+    if ldb.is_registered():
+        entries = getDatabankEntries(ldb.name)
+        local_files, urlnames = entries['path'], entries['download_url']
+        if 'parfunc' in entries:
+            pf_path = [entries['parfunc']]
+    
+    get_pf_files = True
+    if ldb.is_registered() and not radis.config["ALLOW_OVERWRITE"]:
+        error = False
+        if cache == "regen" or not local_files:
+            error = True
+        files_to_check = local_files + pf_path
+        if ldb.get_missing_files(files_to_check):
+            error = True
+        if not pf_path:
+            get_pf_files = False #assume partition function file is either unavailable or undesired for current species
+        if error:
+            raise Exception('Changes are required to the local database, and hence updating the registered entry, but "ALLOW_OVERWRITE" is False')
+    
+    # Delete files if needed:
+
+    if cache == "regen":
+        ldb.remove_local_files(local_files + pf_path)
+    ldb.check_deprecated_files(
+        ldb.get_existing_files(local_files),
+        auto_remove=True if cache != "force" else False,
+    )
+    
+    get_main_files = True
+
+    if len(local_files) > 1 or len(urlnames) > 1:
+        raise Exception('only 1 database file is expected')
```

Comment:
Add suggestion to the user, such as "delete this folder"

Reply:
The issue shouldn't arise if the db entry was automatically registered by RADIS. This is intended more for a case where the user provides their own custom settings.